### PR TITLE
Check for presence of subscription before unsubscribing

### PR DESCRIPTION
**Pull request by srt32**
Fixes https://app.getsentry.com/hound-1/production/group/60210071/
https://trello.com/c/t0cytZ8d/565-check-for-presence-of-subscription-before-unsubscribing

### CODE REVIEW

File: app/controllers/subscriptions_controller.rb [hunk 1]
```diff
@@ -47,6 +48,15 @@ def delete_subscription
     RepoSubscriber.unsubscribe(repo, repo.subscription.user)
   end
 
+  def check_subscription_presence
+    unless repo.subscription.present?
```
Comment (gylaz) [inline app/controllers/subscriptions_controller.rb]: How about `if repo.subscription.blank?`?

File: spec/controllers/subscriptions_controller_spec.rb [hunk 2]
```diff
@@ -74,38 +74,63 @@
...
+
+  context "when there is a subscription" do
+    it "deletes subscription associated with subscribing user" do
+      token = "usertoken"
+      current_user = create(:user)
+      subscribed_user = create(:user)
+      repo = create(:repo, private: true)
+      create(:membership, repo: repo, user: current_user)
+      create(:subscription, repo: repo, user: subscribed_user)
+      activator = double(:repo_activator, deactivate: true)
```
Comment (gylaz) [inline spec/controllers/subscriptions_controller_spec.rb]: Can we give this double a string name `"RepoActivator"`?

File: spec/controllers/subscriptions_controller_spec.rb [hunk 3]
```diff
@@ -74,38 +74,63 @@
...
-    activator = double(:repo_activator, deactivate: true)
-    allow(RepoActivator).to receive(:new).and_return(activator)
-    allow(RepoSubscriber).to receive(:unsubscribe).and_return(true)
-    stub_sign_in(current_user, token)
+  context "when there is no subscription" do
+    it "returns 409 conflict" do
+      current_user = create(:user)
+      repo = create(:repo, private: true)
+      create(:membership, repo: repo, user: current_user)
+      activator = double(:repo_activator, deactivate: true)
```
Comment (gylaz) [inline spec/controllers/subscriptions_controller_spec.rb]: Sting name for double?

**Comment by gylaz** [comment]
Looks good.

**Comment by srt32** [comment]
Great feedback, @gylaz! Done!

**Pull request merged**